nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EvaluateXQuery.java

```diff
@@ -272,10 +315,9 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 public void process(final InputStream rawIn) throws IOException {
                     try (final InputStream in = new BufferedInputStream(rawIn)) {
                         XQueryEvaluator qe = slashExpression.load();
-                        qe.setSource(new SAXSource(new InputSource(in)));
-                        DocumentBuilderFactory dfactory = DocumentBuilderFactory.newInstance();
-                        dfactory.setNamespaceAware(true);
-                        Document dom = dfactory.newDocumentBuilder().newDocument();
+                        final Document document=docBuilder.parse(in);
+                        qe.setSource(new DOMSource(document));
+                        Document dom = docBuilder.newDocument();
```

Comment:
Is there any performance impact by changing from SAX to DOM to evaluate a large XML document?